python/test_MemberParser.py

```diff
@@ -121,6 +123,44 @@ def test_parse_valid(self):  # pylint: disable=too-many-statements
     self.assertTrue(parsed.is_builtin_array)
     self.assertEqual(parsed.array_type, r'std::uint32_t')
 
+  def test_parse_valid_default_value(self):
+    """Test that member variables can be parsed correctly if they have a user
+    defined default value"""
+    parser = MemberParser()
+
+    parsed = parser.parse(r'int fortyTwo{43} // default values can lie')
+    self.assertEqual(parsed.full_type, r'int')
+    self.assertEqual(parsed.name, r'fortyTwo')
+    self.assertEqual(parsed.description, 'default values can lie')
+    self.assertEqual(parsed.default_val, r'43')
+    self.assertEqual(str(parsed), 'int fortyTwo{43}; ///< default values can lie')
+
+    parsed = parser.parse(r'float f{3.14f}', require_description=False)
+    self.assertEqual(parsed.full_type, 'float')
+    self.assertEqual(parsed.name, 'f')
+    self.assertEqual(parsed.default_val, '3.14f')
+
+    parsed = parser.parse(r'std::array<int, 3> array{1, 2, 3} // arrays can be initialized')
+    self.assertEqual(parsed.full_type, r'std::array<int, 3>')
+    self.assertEqual(parsed.default_val, '1, 2, 3')
+    self.assertEqual(parsed.name, 'array')
+
+    # These are cases where we cannot really decide whether the initialization
+    # is valid just from the member declaration. We let them pass here
+    parsed = parser.parse('nsp::SomeValue val {42} // default values can have space')
+    self.assertEqual(parsed.full_type, 'nsp::SomeValue')
+    self.assertEqual(parsed.name, 'val')
+    self.assertEqual(parsed.default_val, '42')
+    self.assertEqual(parsed.namespace, 'nsp')
+    self.assertEqual(parsed.bare_type, 'SomeValue')
+
+    parsed = parser.parse(r'edm4hep::Vector3d v{1, 2, 3, 4} // for aggregates we do not validate init values')
+    self.assertEqual(parsed.full_type, 'edm4hep::Vector3d')
+    self.assertEqual(parsed.default_val, '1, 2, 3, 4')
+
+    parsed = parser.parse(r'AggType bogusInit{here, we can even put invalid c++}', False)
+    self.assertEqual(parsed.default_val, 'here, we can even put invalid c++')
```

Comment:
Examples of things that we cannot really validate without further context, and that we currently pass through verbatim to the generation. Both of these would lead to compilation errors later.